web/src/buddy_data.js

```diff
@@ -86,7 +107,9 @@ export function compare_function(a, b) {
 
 export function sort_users(user_ids) {
     // TODO sort by unread count first, once we support that
-    user_ids.sort(compare_function);
+    const current_sub = narrow_state.stream_sub();
+    const pm_ids = new Set(narrow_state.pm_ids_string()?.split(","));
```

Comment:
I think `people.user_ids_string_to_ids_array` might be better here -- we want to try to avoid duplicated parsing code to the extent possible because it's hard to check if it's correct.

Reply:
Thanks for pointing me to that! The code is a bit longer now but hopefully easier to read. Let me know if you think there's a nice way to clean it up further.

before: `const pm_ids = new Set(narrow_state.pm_ids_string()?.split(","));`

after: 

```js
    const pm_ids_string = narrow_state.pm_ids_string();
    if (pm_ids_string) {
        const pm_ids = new Set(people.user_ids_string_to_ids_array(pm_ids_string));
        user_ids.sort((a, b) => compare_function(a, b, current_sub, pm_ids));
    }
    return user_ids;
```

Reply:
Thanks for pointing me to that! The code is a bit longer now but hopefully easier to read? I'm not convinced though. Let me know if you think there's a nice way to clean it up further. Maybe `people.user_ids_string_to_ids_array` takes an empty string?

before: `const pm_ids = new Set(narrow_state.pm_ids_string()?.split(","));`

after: 
```js
const pm_ids_string = narrow_state.pm_ids_string();
const pm_ids_list = pm_ids_string ? people.user_ids_string_to_ids_array(pm_ids_string) : [];
const pm_ids_set = new Set(pm_ids_list);
user_ids.sort((a, b) => compare_function(a, b, current_sub, pm_ids_set));
```

Reply:
The resulting code seems OK to me. The other reasonable option would be a more explicit if/else.